plugins/artifacts/config-schema.yml

```diff
@@ -0,0 +1,28 @@
+$schema: http://json-schema.org/draft-04/schema#
+title: config
+description: Artifacts to be published
+type: array
+items:
+  type: object
+  properties:
+    type:
+      title: Upload type
+      description: Artifacts can be either an individual `file` or a `directory` containing potentially multiple files with recursively included subdirectories
+      type: string
+      enum: [file, directory]
+    path:
+      title: Artifact Path
+      description: Filesystem path of artifact
+      type: string
+    name:
+      title: Artifact Name
+      description: This will be the leading path to directories and the full name for files that are uploaded to s3
+      type: string
+    expires:
+      title: Expiry Tate and Time
+      description: Date when artifact should expire must be in the future
+      type: string
+      format: date-time
+
+  # TODO: Should expires be required?
```

Comment:
I would love of expires to be optional... and default to task.expires. That seems sane...